cpp/src/arrow/c/bridge.cc

```diff
@@ -714,11 +713,11 @@ Result<std::pair<std::optional<DeviceAllocationType>, int64_t>> ValidateDeviceIn
   return std::make_pair(device_type, device_id);
 }
 
-Status ExportDeviceArray(const Array& array, RawSyncEvent sync_event,
+Status ExportDeviceArray(const Array& array, std::shared_ptr<DeviceSync>& sync,
```

Comment:
nitpick: since we're copying the `shared_ptr` at line 741 anyway, I think it's technically more efficient to take `sync` by value here and then it can be moved at line 741.

Reply:
Agreed, please take this argument by value
```suggestion
Status ExportDeviceArray(const Array& array, std::shared_ptr<DeviceSync> sync,
```